File: Unit 1 Fundamentals of programming/Fundamentals Topic 6 Files and exception handling/Topic 6 Python Programs/W6 Task 2 binary graphic  file.py
#PP L6 binary graphic file
#and copy it to another file

#set a buffer size so memory does not overflow
#the graphic is 476KB
buffersize = 50000
picturefile = open('waterlilies.jpg','rb')
print('File opened')
newfilename = input("Enter the new file name to copy image to: ")
copiedfile = open(newfilename,'wb')

#read 50000 bytes at a time
buffer = picturefile.read(buffersize)
#print a dot every time 50K bytes are written
while len(buffer):
   copiedfile.write(buffer)
   print('.', end = '')
   buffer = picturefile.read(buffersize)
print()
print('file copied ... look in folder!')


              

File: Unit 1 Fundamentals of programming/Fundamentals Topic 6 Files and exception handling/Topic 6 VB Programs/W6 Task 2 binary graphic file.vb
Module Module1



    Sub Main()
        'PP6 binary graphic file and copy it to another file

        Dim file As String = "U:\waterlilies.jpg"
        Dim file2 As String = "U:\waterlilies2.jpg"
        Dim buffer As Integer = 50000

        'set a buffer size so memory does not overflow
        'the graphic is 476KB

        Dim tempFile = IO.File.Open(file, IO.FileMode.Open)

        ' Adjust array length for VB array declaration. 
        Dim bytes = New Byte(buffer - 1) {}

        While tempFile.Read(bytes, 0, buffer) > 0
            My.Computer.FileSystem.WriteAllBytes(file2, bytes, True)
            Console.Write(".")
        End While

        tempFile.Close()


        MsgBox("OK")

    End Sub





End Module




